src/gromacs/nblib/box.cpp

```diff
@@ -72,14 +74,16 @@ Box::Box(real x, real y, real z)
     box_(XX, XX) = x;
     box_(YY, YY) = y;
     box_(ZZ, ZZ) = z;
+
+    gmx::fillLegacyMatrix(box_, legacyMatrix);
```

Comment:
The legacy matrix is a c++ array. So one can return a reference to an array, or copy it in-place. It cannot be returned like STL containers. Hence, I'd fill up such an array in-place and access it when needed. I should add const protections to it.

Reply:
Fair enough. Since this is the case I think that we should not add anything to the Box class. We will have to call fillLegacyMatrix every time we need to copy anyway so there is no savings in terms of coding or increase in clarity.

Reply:
Not true. We fill the legacy matrix at the time of construction. Then this legacy matrix is used for any calculations during setup.

Reply:
Ack